tools/pythonpkg/src/arrow_array_stream.cpp

```diff
@@ -35,20 +39,34 @@ unique_ptr<ArrowArrayStreamWrapper> PythonTableArrowArrayStreamFactory::Produce(
 			}
 		}
 
+	} else if (py_object_type == "RecordBatchReader") {
+		// py::object arrow_batch_scanner = py::module_::import("pyarrow.dataset").attr("Scanner").attr("from_batches");
+		// scanner = arrow_batch_scanner(arrow_obj_handle);
+		// auto record_batches = scanner.attr("to_reader")();
+		auto res = make_unique<ArrowArrayStreamWrapper>();
+		auto export_to_c = arrow_obj_handle.attr("_export_to_c");
+		export_to_c((uint64_t)&res->arrow_array_stream);
+		return res;
```

Comment:
I think you can just add ```|| py_object_type == "RecordBatchReader")``` to the if clause code above, the recordbatchreader should follow the same path as the table.

Reply:
The arrow dataset does not seem to accept a reader as input. It says it should in the docs, but it's not implemented in the [code](https://github.com/apache/arrow/blob/master/python/pyarrow/dataset.py#L668) at least as far as I can see.

Reply:
So the scanner that is created from_batches is per definition single use and we cannot make a dataset from the reader. Even if we pass scanner coming from a reader to the dataset it wont work as after any query the generator will be consumed. I guess one can create a view and then create an actual table to hack around this a bit, but that is not optimal. Or reread from the Flight Server.

Reply:
Hm, I see
I can have a look on it next week!

Reply:
I discussed this a bit on the arrow issues [boards](https://github.com/apache/arrow/issues/12667) and this is the correct behaviour basically.